src/arkode/arkode_arkstep_nls.c

```diff
@@ -427,10 +433,13 @@ int arkStep_NlsLSolve(N_Vector b, void* arkode_mem)
   retval = arkStep_AccessStepMem(arkode_mem, "arkStep_NlsLSolve",
                                  &ark_mem, &step_mem);
   if (retval != ARK_SUCCESS)  return(retval);
+  
+  SUNAssignSUNCTX(ark_mem->sunctx);
 
   /* retrieve nonlinear solver iteration from module */
   retval = SUNNonlinSolGetCurIter(step_mem->NLS, &nonlin_iter);
-  if (retval != SUN_NLS_SUCCESS)
+  SUNCheckCallNoRet(retval);
+  if (retval != SUN_SUCCESS)
     return(ARK_NLS_OP_ERR);
```

Comment:
Since this function now returns a `SUNNlsStatus`, should the returned values correspond to SUNDIALS-wide error constants (instead of package-specific ones like `ARK_NLS_OP_ERR`)?